### PR TITLE
More work on the macros chapters

### PR DESCRIPTION
**Pull request by kmcallister**
r? @steveklabnik

### CODE REVIEW

File: src/doc/trpl/macros.md [hunk 1]
```diff
@@ -226,6 +224,10 @@ That's most of the matcher syntax. These examples use `$(...)*`, which is a
 more" match. Both forms optionally include a separator, which can be any token
 except `+` or `*`.
 
+This system is based on
+"[Macro-by-Example](http://www.cs.indiana.edu/ftp/techreports/TR206.pdf)"
+(PDF link).
```
Comment (steveklabnik) [inline src/doc/trpl/macros.md]: hmmm, we have no current convention about this. it's fine, just something to think about

**Comment by kmcallister** [comment]
Rebased.

**Pull request merged**